collector/src/lib.rs

```diff
@@ -334,59 +275,190 @@ where
 {
 }
 
+/// Builder for configuring a [`Collector`].
+pub struct CollectorBuilder<V: vdaf::Collector> {
+    /// Unique identifier for the task.
+    task_id: TaskId,
+    /// The base URL of the leader's aggregator API endpoints.
+    leader_endpoint: Url,
+    /// The authentication information needed to communicate with the leader aggregator.
+    authentication: AuthenticationToken,
+    /// HPKE keypair used for decryption of aggregate shares.
+    hpke_keypair: HpkeKeypair,
+    /// An implementation of the task's VDAF.
+    vdaf: V,
+
+    /// HTTPS client.
+    http_client: Option<reqwest::Client>,
+    /// Parameters to use when retrying HTTP requests.
+    http_request_retry_parameters: ExponentialBackoff,
+    /// Parameters to use when waiting for a collection job to be processed.
+    collect_poll_wait_parameters: ExponentialBackoff,
+}
+
+impl<V: vdaf::Collector> CollectorBuilder<V> {
+    /// Construct a [`CollectorBuilder`] from required DAP task parameters and an implementation of
+    /// the task's VDAF.
+    pub fn new(
+        task_id: TaskId,
+        leader_endpoint: Url,
+        authentication: AuthenticationToken,
+        hpke_keypair: HpkeKeypair,
+        vdaf: V,
+    ) -> Self {
+        Self {
+            task_id,
+            leader_endpoint,
+            authentication,
+            hpke_keypair,
+            vdaf,
+            http_client: None,
+            http_request_retry_parameters: http_request_exponential_backoff(),
+            collect_poll_wait_parameters: ExponentialBackoff {
+                initial_interval: StdDuration::from_secs(15),
+                max_interval: StdDuration::from_secs(300),
+                multiplier: 1.2,
+                max_elapsed_time: None,
+                ..Default::default()
+            },
+        }
+    }
+
+    /// Finalize construction of a [`Collector`].
+    pub fn build(self) -> Result<Collector<V>, Error> {
+        let http_client = if let Some(http_client) = self.http_client {
+            http_client
+        } else {
+            default_http_client()?
+        };
```

Comment:
Q: Is having the default inserted here just so that `new()` isn't fallible?

Reply:
It was moreso to avoid setting up and tearing down synchronization primitives, etc. for the connection pool if a separate client is supplied, and the default goes unused.